iroh/src/metrics.rs

```diff
@@ -1,7 +1,46 @@
 //! Co-locating all of the iroh metrics structs
+use std::sync::Arc;
+
+use iroh_metrics::{MetricsGroup, MetricsGroupSet};
 #[cfg(feature = "test-utils")]
 pub use iroh_relay::server::Metrics as RelayMetrics;
 #[cfg(not(wasm_browser))]
 pub use portmapper::Metrics as PortmapMetrics;
 
 pub use crate::{magicsock::Metrics as MagicsockMetrics, net_report::Metrics as NetReportMetrics};
+
+/// Metrics collected by an [`crate::endpoint::Endpoint`].
+///
+/// See [`crate::endpoint::Endpoint::metrics`] for details.
+#[derive(Default, Debug, Clone)]
+pub struct EndpointMetrics {
```

Comment:
we probably should mark all the metrics structs as `non_exhaustive`

Reply:
Done